### ISSUE TITLE
Shell commands opening VIM results in a freeze

### ISSUE THREAD

**Issue by Casebash** [ipython/ipython#10975]
Some shell commands open up a VIM console. When this happens Jupyter QtConsole freezes.

Easiest way to reproduce is to run !deepdive do init/app

**System Details**

$ jupyter qtconsole --version
4.3.1

{'commit_hash': 'd86648c5d',
 'commit_source': 'installation',
 'default_encoding': 'UTF-8',
 'ipython_path': '/Users/chris/anaconda3/lib/python3.6/site-packages/IPython',
 'ipython_version': '6.1.0',
 'os_name': 'posix',
 'platform': 'Darwin-17.3.0-x86_64-i386-64bit',
 'sys_executable': '/Users/chris/anaconda3/bin/python',
 'sys_platform': 'darwin',
 'sys_version': '3.6.3 |Anaconda, Inc.| (default, Oct  6 2017, 12:04:38) \n'
                '[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]'}

**Comment by takluyver** [comment]
Yup, any shell commands which expect user interaction get stuck. We've been aware of this pretty much since the Qt console was made, but we don't know of a good fix which doesn't make things worse in other cases.

**Comment by ma-public** [comment]
Its 3 years and have not yet fixed. We should wait another 10 years i think
